internal/providers/pluginfw/converters/go_to_tf.go

```diff
@@ -151,9 +166,19 @@ func goSdkToTfSdkSingleField(ctx context.Context, srcField reflect.Value, destFi
 			// Skip zeros
 			return nil
 		}
-		// resolve the nested struct by recursively calling the function
-		if GoSdkToTfSdkStruct(ctx, srcFieldValue, destField.Addr().Interface()).HasError() {
-			panic(fmt.Sprintf("%s. %s", goSdkToTfSdkStructConversionFailureMessage, common.TerraformBugErrorMessage))
+		if destField.Kind() == reflect.Slice {
+			// allocate a slice first
+			destSlice := reflect.MakeSlice(destField.Type(), 1, 1)
+			// resolve the nested struct by recursively calling the function
+			if GoSdkToTfSdkStruct(ctx, srcFieldValue, destSlice.Index(0).Addr().Interface()).HasError() {
+				panic(fmt.Sprintf("%s. %s", goSdkToTfSdkStructConversionFailureMessage, common.TerraformBugErrorMessage))
+			}
+			destField.Set(destSlice)
+		} else {
+			// resolve the nested struct by recursively calling the function
+			if GoSdkToTfSdkStruct(ctx, srcFieldValue, destField.Addr().Interface()).HasError() {
+				panic(fmt.Sprintf("%s. %s", goSdkToTfSdkStructConversionFailureMessage, common.TerraformBugErrorMessage))
+			}
 		}
```

Comment:
This is parallel to the pointer case and can be structured as such.
```suggestion
		var dest any
		if destField.Kind() == reflect.Slice {
			// allocate a slice first
			destSlice := reflect.MakeSlice(destField.Type(), 1, 1)
			destField.Set(destSlice)
			dest = destSlice.Index(0).Addr().Interface()
		} else {
			dest = destField.Addr().Interface()
		}
		// resolve the nested struct by recursively calling the function
		if GoSdkToTfSdkStruct(ctx, srcFieldValue, dest).HasError() {
			panic(fmt.Sprintf("%s. %s", goSdkToTfSdkStructConversionFailureMessage, common.TerraformBugErrorMessage))
		}
```